pkg/cdi/cache.go

```diff
@@ -162,6 +219,11 @@ func (c *Cache) InjectDevices(ociSpec *oci.Spec, devices ...string) ([]string, e
 	c.Lock()
 	defer c.Unlock()
 
+	// in self-refresh mode trigger a refresh here if a Spec dir appears
+	if c.selfRefresh && c.watch.update() {
```

Comment:
It seems strange to me that this is required here (or only here). Should all queries to the cache not require a refresh?

I would expect the watcher to trigger the refresh on updates to ensure that the cache state is always up to date.

Reply:
Yes, in principle all of them should have this same snippet to guarantee 100% correct, up-to-date results. This is the only place though which is currently critical from the point of view of runtimes. I can plant it to the rest of the places, too.

Reply:
Updated accordingly.